### PR TITLE
[Release] Twitter App

### PR DESCRIPTION
**Pull request by syumet**
- [Java HTTP & JSON](https://www.notion.so/38ed2bcfc93b421b92de84e5d6f728c3?v=08ebea1ef0a4457197e2910f2d76bcd2&p=101b064340794570ac5359b2f16c8a4b)

- [DAO Implementation](https://www.notion.so/38ed2bcfc93b421b92de84e5d6f728c3?v=08ebea1ef0a4457197e2910f2d76bcd2&p=fa4671307809447f938463af351b5b2b)

- [Service Implementation](https://www.notion.so/38ed2bcfc93b421b92de84e5d6f728c3?v=08ebea1ef0a4457197e2910f2d76bcd2&p=3a1e491f41034e56bb4bd295003d0834)

- [Controller & App Implementation](https://www.notion.so/38ed2bcfc93b421b92de84e5d6f728c3?v=08ebea1ef0a4457197e2910f2d76bcd2&p=775b8ea595a1440a9c87771429a9a0f1)

- [Spring](https://www.notion.so/38ed2bcfc93b421b92de84e5d6f728c3?v=08ebea1ef0a4457197e2910f2d76bcd2&p=acf5ce49d6794d9d9ab28ae28754fc8c)

- [Twitter README.md](https://www.notion.so/38ed2bcfc93b421b92de84e5d6f728c3?v=08ebea1ef0a4457197e2910f2d76bcd2&p=bef1ef5b01654c8687ca52194fc135d1)

Note: Release PR of [JDBC](https://github.com/jarviscanada/jarvis_data_eng_steve/pull/22) project has been merged.

### CODE REVIEW

**Comment by syumet** [comment]
Some file permission may be changed accidentally, so the commits from JDBC are still here... not sure how to prevent this

File: core_java/twitter/src/main/java/ca/jrvs/apps/twitter/controller/TwitterController.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+      );
+    }
+
+    Double lat, lon;
+    try {
+      lat = Double.parseDouble(coordinateArray[0]);
+      lon = Double.parseDouble(coordinateArray[1]);
+    } catch (NumberFormatException e) {
+      throw new IllegalArgumentException(
+          e.getMessage() + "Invalid location format\n" + POST_USAGE,
```
Comment (jarviscanada) [inline core_java/twitter/src/main/java/ca/jrvs/apps/twitter/controller/TwitterController.java]: you don't need to append `e.getMessage()` as this message will appear in the stack trace anyway

File: core_java/twitter/src/main/java/ca/jrvs/apps/twitter/utils/TweetUtil.java [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+  public static final String FIELD_SEP = ",";
+  public static final String CMD_POST = "post";
+  public static final String CMD_SHOW = "show";
+  public static final String CMD_DEL = "delete";
+  private static final Pattern ID_PATTERN = Pattern.compile("^[0-9]+$");
+
+  private static final String[] ALL_FIELDS;
+  private static final HashMap<String, Method> SETTER_TABLE;
+
+  static {
```
Comment (jarviscanada) [inline core_java/twitter/src/main/java/ca/jrvs/apps/twitter/utils/TweetUtil.java:35]: every interesting solution. i like it

File: core_java/twitter/src/main/java/ca/jrvs/apps/twitter/dao/helper/TwitterHttpHelper.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+//  /**
+//   * Constructor setup dependencies using secrets
+//   */
+//  public TwitterHttpHelper(String consumerKey, String consumerSecret,
+//      String accessToken, String tokenSecret) {
+//    this.consumer = new CommonsHttpOAuthConsumer(consumerKey, consumerSecret);
+//    this.consumer.setTokenWithSecret(accessToken, tokenSecret);
+//    this.httpClient = HttpClientBuilder.create().build();
+//  }
```
Comment (jarviscanada) [inline core_java/twitter/src/main/java/ca/jrvs/apps/twitter/dao/helper/TwitterHttpHelper.java]: clean up unused code

**Review by jarviscanada — APPROVED** [review]
well done

**Pull request merged**